File: Data Structures & Algorithms/maximum-product-subarray/submission-8.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums: number[]): number {
        if(nums.length===1) return nums[0]
        let maxProduct=-Infinity;
        let prev=1
        for(let i=0;i<nums.length;i++){
            for(let j=i;j<nums.length;j++){
            if(i===j)prev=nums[i]*1;
            else{
                prev*=nums[j]
            }
            maxProduct=Math.max(maxProduct, prev)
            }
        }
        return maxProduct;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-8.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        if not nums:
            return 0

        def check(nums: List[int]) -> int:
            if not nums:
                return 0

            p1 = 0
            p2 = 0
            for num in nums:
                temp = p2
                p2 = max(p2, p1 + num)
                p1 = temp
            return p2

        return max(check(nums[1:]), check(nums[0 : len(nums) - 1]))
